turbo/jsonrpc/eth_api.go

```diff
@@ -335,32 +335,34 @@ func (api *BaseAPI) blockNumberFromBlockNumberOrHash(tx kv.Tx, bnh *rpc.BlockNum
 // APIImpl is implementation of the EthAPI interface based on remote Db access
 type APIImpl struct {
 	*BaseAPI
-	ethBackend      rpchelper.ApiBackend
-	txPool          txpool.TxpoolClient
-	mining          txpool.MiningClient
-	gasCache        *GasPriceCache
-	db              kv.RoDB
-	GasCap          uint64
-	ReturnDataLimit int
-	logger          log.Logger
+	ethBackend                  rpchelper.ApiBackend
+	txPool                      txpool.TxpoolClient
+	mining                      txpool.MiningClient
+	gasCache                    *GasPriceCache
+	db                          kv.RoDB
+	GasCap                      uint64
+	ReturnDataLimit             int
+	MaxGetProofRewindBlockCount int
+	logger                      log.Logger
 }
 
 // NewEthAPI returns APIImpl instance
-func NewEthAPI(base *BaseAPI, db kv.RoDB, eth rpchelper.ApiBackend, txPool txpool.TxpoolClient, mining txpool.MiningClient, gascap uint64, returnDataLimit int, logger log.Logger) *APIImpl {
+func NewEthAPI(base *BaseAPI, db kv.RoDB, eth rpchelper.ApiBackend, txPool txpool.TxpoolClient, mining txpool.MiningClient, gascap uint64, returnDataLimit int, maxGetProofRewindBlockCount int, logger log.Logger) *APIImpl {
```

Comment:
It's not exactly following the pattern that they've established, but... I think it would make our lives much easier/nicer if we were to simply not add this parameter to `NewEthAPI` and instead where necessary do something like:

```
api := NewEthAPI( ... )
api.MaxProofRewindBlockCount = cfg.MaxProofRwindBlockCount
```

It's not ideal, but, by avoiding changing the function signature, we'll create many fewer conflicts down the road, as any modified tests or new tests referencing this old signature will give us problems.

Reply:
touche!

Reply:
If we plan to push upstream, then modifying the constructor is probably fine.  My concern was that we were introducing this into our repo first and that the churn related to maintaining it would be high.

FWIW, I think the constructor itself is not particularly idiomatic.  If instead, things were simply initialized as a struct, allowing zero defaults, it would clean up the code quite a bit, no more NewEthAPI(..., nil, nil, nil, ...) that we see all over the place.  But, obviously such a deviation would need to be accepted upstream first (so I'm not advocating for this refactoring as part of the charge)

Reply:
okay, then I will reopen this PR and pull the changes and adapt, then push upstream! 

whether we decide to refactor and cleanup `NewEthAPI` can either come from us or we can open up an issue within their repo. Thanks!